### PR TITLE
Adds Nightwatch test that can be run under Travis to test the installer

### PR DESCRIPTION
**Pull request by justafish**
Task: #123

* [ ] Ready for review
* [ ] Ready for merge

This PR adds a Travis configuration which will install Contenta and then run a simple Nightwatch test to check it worked.

### Notes

This splits up the composer script to install and start contenta

### Test Instructions

See if Travis goes green!

### CODE REVIEW

**Comment by e0ipso** [comment]
Do we want a separate CI for this?

**Comment by dawehner** [comment]
I think what matters is that it works. If we want, we can always refactor the CI system, as long the tests are actually running I'm cool.

**Comment by e0ipso** [comment]
> I think what matters is that it works. If we want, we can always refactor the CI system, as long the tests are actually running I'm cool.

Yeah, totally. I was thinking that given that it is probably less work to do it in Circle CI (given that we are already installing Drupal in a Pantheon multidev, per PR), we'd want to go with that.

It may even fix the currently broken build.

File: installer.sh [hunk 1]
```diff
@@ -4,17 +4,11 @@ original_wd=$(pwd)
 mytmpdir=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')
 git clone --depth=1 git@github.com:contentacms/contenta_jsonapi.git $mytmpdir
 cd $mytmpdir
-read -n 1 -p "Do you wish to install Contenta CMS (y/n)? " answer
```
Comment (e0ipso) [inline installer.sh:7]: Let's not remove this, please. It's good practice to have people re-confirm when they are running an unknown script from the Internet.

File: scripts/development/start-contenta_jsonapi.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+export $(cat .env | xargs)
+if [ -e x.txt ]
```
Comment (e0ipso) [inline scripts/development/start-contenta_jsonapi.sh:4]: What is `x.txt`?

File: scripts/development/build-contenta_jsonapi.sh [hunk 3]
```diff
@@ -72,16 +72,19 @@ echo -e "\t********************************"
...
-echo " (4/4) Initializing local PHP server "
-echo "-------------------------------------"
-echo -e "${FG_C}${WBG_C} INFO ${NO_C} Server started. Use Ctrl+C to stop it."
-echo -e "${FG_C}${WBG_C} INFO ${NO_C} If you want to start the server manually use:\n\t$DRUSH runserver"
-# Start the server in the background to be able to get the one time login link.
-$DRUSH runserver
+while getopts ":s" opt; do
+  case $opt in
+    s)
+
```
Comment (e0ipso) [inline scripts/development/build-contenta_jsonapi.sh:84]: Did you mean to add `./start-contenta_jsonapi.sh` here?

We should document the `-s` option somewhere. I think that making the server start optional requires adding some sort of user help for the script.

File: scripts/development/build-contenta_jsonapi.sh [hunk 4]
```diff
@@ -72,16 +72,19 @@ echo -e "\t********************************"
...
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
```
Comment (e0ipso) [inline scripts/development/build-contenta_jsonapi.sh:87]: Let's either:
  1. Ignore the fact that an unknown option was passed.
  1. Exit with a non-zero value to denote error.

I don't have any opinions in which way to go.

**Comment by e0ipso** [comment]
For curiosity, is there any benefit of using Nightwatch instead of Behat here? Probably JS interactions?

**Comment by dawehner** [comment]
> For curiosity, is there any benefit of using Nightwatch instead of Behat here? Probably JS interactions?

For me at least behat is often abused for pure testing purposes, even well, the real value is in gherkin, aka. finding a common language with the client. If you don't have the need for that, writing the step definitions and maintaining them is a pain in the ass. I think having a pure code based testing framework is the better choice for our little project here.

**Comment by e0ipso** [comment]
See https://github.com/contentacms/contenta_jsonapi/pull/71 for an alternative that does not require a new CI.

**Pull request closed without merging**